app/app/index.js

```diff
@@ -1,96 +1,156 @@
+        },
+      )
+
+      const user = await response.json()
+      setUserInfo(user)
+    } catch (error) {
+      console.log(1)
+    }
+  }
+
+  return (
+    <SafeAreaView style={globalStyles.container}>
+      <View style={globalStyles.appHeader}>
+        <Text style={globalStyles.appMainTitle}>FOOD4THOUGHT</Text>
       </View>
 
-      <View style = {globalStyles.appBody}>
-        <Text style = {globalStyles.appBodyFont}> Say hello to endless culinary possibilities! </Text>
+      <View style={globalStyles.appBody}>
+        <Text style={globalStyles.appBodyFont}>
+          {' '}
+          Say hello to endless culinary possibilities!{' '}
+        </Text>
       </View>
 
-      <View style = {globalStyles.appLogin}>
-        <Text style = {[globalStyles.appBodyFont, 
-          {fontSize: 35, fontWeight: '700', alignSelf: 'center', marginVertical: 15,}]}>
+      <View style={globalStyles.appLogin}>
+        <Text
+          style={[
+            globalStyles.appBodyFont,
+            {
+              fontSize: 35,
+              fontWeight: '700',
+              alignSelf: 'center',
+              marginVertical: 15,
+            },
+          ]}
+        >
           Login
-          </Text>
-        <TextInput 
-          style = {globalStyles.userInputs} 
-          autoCapitalize='none' 
-          keyboardType = 'email-address' 
-          placeholder='Email' 
+        </Text>
+        <TextInput
+          style={globalStyles.userInputs}
+          autoCapitalize="none"
+          keyboardType="email-address"
+          placeholder="Email"
           value={email}
-          onChangeText={text => setEmail(text)} 
-          />
-        <TextInput 
-          style = {globalStyles.userInputs} 
-          blurOnSubmit = {true}  
-          placeholder='Password'
+          onChangeText={(text) => setEmail(text)}
+        />
+        <TextInput
+          style={globalStyles.userInputs}
+          blurOnSubmit={true}
+          placeholder="Password"
           value={password}
-          onChangeText={text => setPassword(text)}
+          onChangeText={(text) => setPassword(text)}
           secureTextEntry
-          />
+        />
 
-        <FlatButton text={'Sign In'} onPress={handleLogin} invert={'n'}/>
-        <Link href="/password" style = {{
-            borderRadius: 15,
-            padding: 12, 
-            margin: 10, 
-            width: 280,
-            color: 'black',
-            fontWeight: 'bold',
-            fontSize: 15,
-            textAlign: 'center',
-        }}>
-          Forget Password
+        <FlatButton
+          text={'Sign In'}
+          onPress={handleLogin}
+          invert={'n'}
+          disabled={false}
+        />
+        <FlatButton
+          text={'Sign In with Google'}
+          onPress={() => promptAsync()} // you need to do "lazily" call promptAsync. Previously it was onPress={promptAsync} which means that everytime you rendered the app, it always invoke the auth session 
```

Comment:
@chenxk619 @BenjoBurger see this for the fix, thanks!

Reply:
Every time you have `on...` its almost always a lazy call to some function 😄

Reply:
thank you!!